lib/compute/agent-nodes.ts

```diff
@@ -281,4 +281,15 @@ export class AgentNodes {
       remoteFs: '/home/ec2-user',
     };
   }
+
+  public getRequiredAgentNodes(type: string): any[] {
+    return Object.keys(this)
+      .filter((key) => {
```

Comment:
We are doing similar thing with _jenkinsType_ variable.
If _jenkinsType_ value is not passed we assume it's default(this option is mostly used for testing), in all other cases we retrieve the agent nodes list with the value passed in _jenkinsType_(Ex: benchmark, gradle). But for BTR we need to exclude the agentnode names having the values ('default', 'benchmark', 'gradle').

The Drawback is agentnodes used by BTR instance cannot be differentiated from their names like we do with benchmark and gradle. We can think of updating agentnodes names used by BTR with a key_name(RELEASE) if it's straightforward.

Reply:
I think since we have a defined number of instance type, better have a check, may be enums, to verify if a valid instance type name is passed. 
If invalid name is passed then throw an error.

Reply:
Agreed. Have enums for instance types. Also what is BTR? 😅

Reply:
Yes Sayali